sound/soc/soc-core.c

```diff
@@ -1139,6 +1139,9 @@ static int snd_soc_compensate_channel_connection_map(struct snd_soc_card *card,
 void snd_soc_remove_pcm_runtime(struct snd_soc_card *card,
 				struct snd_soc_pcm_runtime *rtd)
 {
+	if (!rtd)
+		return;
+
```

Comment:
Is this addressing same case as the first patch? Not saying this is wrong, but just trying to understand the series.

Reply:
Yes, I think the only call site that can result to this is from soc_tplg_remove_link()
I was not sure where to fix this, in soc-topology.c or in soc-core.c.

I can drop the patch for soc-core.c